test/unit/transactionsTest.js

```diff
@@ -47,3 +49,43 @@ describe('calculateTransactionBodiesByteLength()', () => {
     lengthObj.length.should.be.exactly(0)
   })
 })
+
+describe('*createRerunTasks', () => {
+  const transaction = Object.freeze({
+    status: 'Failed',
+    request: {
+      timestamp: new Date().toISOString()
+    },
+    updatedBy: {
+      id: new ObjectId(),
+      name: 'Test'
+    }
+  })
+  const userEmail = 'example@gmail.com'
+
+  beforeEach(async () => {
+    await TransactionModel.deleteMany({})
+    await TaskModel.deleteMany({})
+  })
+
+  afterEach(async () => {
+    await TransactionModel.deleteMany({})
+    await TaskModel.deleteMany({})
+  })
+
+  it('should create rerun task', async () => {
+    await TransactionModel(transaction).save()
+    await transactions.createRerunTasks({}, 1, userEmail, 0, 0, 'Paused', 1)
+    const tasks = await TaskModel.find()
+    tasks.length.should.be.exactly(1)
+  })
+
+  it('should create multiple rerun tasks', async () => {
+    await TransactionModel(transaction).save()
+    await TransactionModel(transaction).save()
```

Comment:
Duplicated line ?

Reply:
It not duplication as such. We actually want to create two transactions here